README.md

```diff
@@ -1 +1,68 @@
-# foundry-multichain-deploy
\ No newline at end of file
+# foundry-multichain-deploy
+
+Provides `foundry` tooling for the multichain deployment contract built atop of
+Sygma. See
+[ChainSafe/hardhat-plugin-multichain-deploy]("https://github.com/ChainSafe/hardhat-plugin-multichain-deploy")
+for more details.
+
+## Usage
+
+The `CrosschainDeployScript` contract is a foundry "script", which means that it
+is not really deployed onto the blockchain. It provides a few helper methods
+that make it easier to deal with the `CrosschainDeployAdapter` from the hardhat
+repository.
+
+To use it, first import the `CrosschainDeployScript`.
+
+```solidity
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity 0.8.20
+
+contract SampleContract {
+    function deployMultichain public payable() {
+        // Remember that forge "builds" the contracts and stores them and their ABI in the root level of `out` so you'd just need to use the contract file name and the contract name and forge gets it from the ABI.
+        CrosschainDeployScript crosschainDeployScript = new CrosschainDeployScript("SimpleContract.sol:SimpleContract");
```

Comment:
shouldn't it make more sense for SampleContract to "extend" CrosschainDeployScript instead?

Reply:
But foundry recommends deploying using Script: https://book.getfoundry.sh/tutorials/solidity-scripting#writing-the-script
we don't want CrosschainDeployScript to end up being deployed (just ran locally) so when user is scripting deployment he would create DeployMyContract.sol which will extend ˙Script` and `CrosschainDeployScript` and script deployment

Reply:
I'm just disagreen with the sample here that instantiates CrosschainDeployScript instead of extending it